src/main/java/Event/Deadline.java

```diff
@@ -0,0 +1,16 @@
+package Event;
+
+public class Deadline extends Task {
+    protected String by;
+    public Deadline(String input) {
+        super(input);
+        String[] parts = input.split(" /by ", 2);
+        this.description = parts[0].substring(9).trim(); //Remove deadline command and trim spaces.
+        this.by = parts[1];
+        this.eventType = "[D]";
```

Comment:
In general, you could refactor string literals (not restricted to this class/file) to adhere to this code quality [guideline](https://nus-cs2113-ay2324s2.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-magic-numbers)